### PR TITLE
Additional pod handler race

### PR DESCRIPTION
**Pull request by trozet**
Fixes another case where update pod and adding pods to retry could race. Also a trivial fix to skipping the pod when it is not scheduled.

### CODE REVIEW

File: go-controller/pkg/ovn/ovn.go [hunk 1]
```diff
@@ -438,9 +440,12 @@ func (oc *Controller) iterateRetryPods(updateAll bool) {
 	defer oc.retryPodsLock.Unlock()
 	now := time.Now()
 	for uid, podEntry := range oc.retryPods {
+		if podEntry.ignore {
```
Comment (abhat) [inline go-controller/pkg/ovn/ovn.go:443]: who sets it back to to ignore = false?

Comment (trozet) [reply go-controller/pkg/ovn/ovn.go:443]: oops, i missed doing that in update, thanks

**Comment by abhat** [comment]
LGTM

**Comment by dcbw** [comment]
@trozet should we have the pod AddFunc() toss an 'ignore' entry for the pod when it starts processing, and unSkipRetryPod() if it fails? Otherwise I think we could have the WatchNodes() add all the pods to the retry list, request a sync, and the retry list starts going through all the pods. At the same time we could get the Add event and start processing pods there too.

That could block the AddFunc() on the retryPodsLock while iterateRetryPods() is happening though. Not sure how else to get around the issue unless we fundamentally rework how pods are handled (which we definitely should).

**Comment by trozet** [comment]
yes good catch @dcbw

File: go-controller/pkg/ovn/ovn.go [hunk 2]
```diff
@@ -469,19 +474,53 @@ func (oc *Controller) checkAndDeleteRetryPod(uid types.UID) bool {
...
-// addRetryPod tracks a failed pod to retry later
-func (oc *Controller) addRetryPod(pod *kapi.Pod) {
+// checkAndSkipRetryPod sets a specific entry from the map to be ignored for subsequent retries
+// if it existed, returns true
+func (oc *Controller) checkAndSkipRetryPod(uid types.UID) bool {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	if entry, ok := oc.retryPods[uid]; ok {
+		entry.ignore = true
+		oc.retryPods[uid] = entry
```
Comment (dcbw) [inline go-controller/pkg/ovn/ovn.go:484]: Maybe we should make retryPods hold pointers to retryEntry; but that can be later.

Comment (trozet) [reply go-controller/pkg/ovn/ovn.go:484]: if you like I can do in this PR. Are there any other comments besides this?

Comment (dcbw) [reply go-controller/pkg/ovn/ovn.go:484]: @trozet Nope.

**Comment by trozet** [comment]
@dcbw updated to use pointers in the map, PTAL

**Pull request merged**